pkgs/tools/misc/rmate-sh/default.nix

```diff
@@ -0,0 +1,47 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, patsh
+}:
+
+stdenv.mkDerivation rec {
+  pname = "rmate";
+  version = "1.0.2";
+
+  src = fetchFromGitHub {
+    owner = "aurora";
+    repo = pname;
+    rev = "refs/tags/v${version}";
+    hash = "sha256-fmK6h9bqZ0zO3HWfZvPdYuZ6i/0HZ1CA3FUnkS+E9ns=";
+  };
+
+  buildPhase = ''
+    runHook preBuild
+
+    ${patsh}/bin/patsh -f rmate -s ${builtins.storeDir}
```

Comment:
patsh should be added to nativeBuildInputs, that way cross compilation would automatically work.